packages/astro/src/core/build/internal.ts

```diff
@@ -221,16 +216,77 @@ export function* getPageDatasByClientOnlyID(
 	}
 }
 
-export function getPageDataByComponent(
+/**
+ * From its route and component, get the page data from the build internals.
+ * @param internals Build Internals with all the pages
+ * @param route The route of the page, used to identify the page
+ * @param component The component of the page, used to identify the page
+ */
+export function getPageData(
 	internals: BuildInternals,
+	route: string, 
 	component: string
 ): PageBuildData | undefined {
-	if (internals.pagesByComponent.has(component)) {
-		return internals.pagesByComponent.get(component);
-	}
+	let pageData = internals.pagesByKeys.get(makePageDataKey(route, component));
+	if (pageData) { return pageData;}
 	return undefined;
 }
 
+/**
+ * Get all pages datas from the build internals, using a specific component.
+ * @param internals Build Internals with all the pages
+ * @param component path to the component, used to identify related pages
+ */
+export function getPagesDatasByComponent(
+	internals: BuildInternals,
+	component: string
+): PageBuildData[] {
+	const pageDatas: PageBuildData[] = [];
+		internals.pagesByKeys.forEach((pageData) => {
+			if (component === pageData.component) pageDatas.push(pageData);
+	})
+	return pageDatas;
+}
+
+/**
+ * TODO: This function is used to avoid breaking changes in the Integrations API.
+ * Should be removed in the future (in Astro 5 ?).
```

Comment:
nit: A `TODO` shouldn't be part of the JSDoc, we should move it in a simple `//` comment. Also, would you mind adding some context? Like, unfortunately, input types and output types are the same, so we should explain **what's the breaking** and how integrations will be affected in Astro 5.

The reason I am asking this is because if we decided to go ahead with the breaking change, we need to provide a migration path to the integrations, and if you're not around, I would like to have this information stored in a comment so we can copy it and give it to downstream users.

Reply:
I don't see the point of putting it in a simple comment rather than a JSdoc? The entire function is affected.

But I've added some context about the old behavior, and why this feature is useful :
```ts
/**
 * TODO: This function is used to avoid breaking changes in the Integrations API.
 * Should be removed in the future (in Astro 5 ?).
 * Parse internals.pagesByKeys to get the page data with the public key.
 * If the page component is unique -> the public key is the component.
 * This match the old behavior of the Integrations API, where pages were identified by the component.
 * If the page component is shared -> the public key is the internal key.
 * This should be the new behavior, since we identify pages by their internal key now.
 * Until then, this is not a breaking change, because a shared entrypoint was not supported before.
 * @param pagesByKeys A map of all page data by their internal key
 */
export function getPageDatasWithPublicKey(pagesByKeys: Map<string, PageBuildData>): Map<string, PageBuildData> {
```

Reply:
>I don't see the point of putting it in a simple comment rather than a JSdoc?

IDEs will render JSDoc, and the `TODO` will be part of the documentation of the function. `TODO` aren't documentation, but internal things.

Reply:
This doesn't really matter here, the entire file (and this function) is internal (as its name suggest). If anything, I'd argue that the `TODO` is part of the documentation of the function, as it signifies that it is temporary and shouldn't be used without considerations.

There's already a few places in Astro with this, in similar contexts (and not, I actually added a TODO to a user-facing method by accident, looking at the code, ha)

Reply:
I'm not an Astro maintainer, so perhaps I don't fully understand your documentation culture. Nevertheless, I am surprised to read that _« [internals documentation] needs the same treatment as public APIs »_ when this part of the code had so little documentation before I contributed, especially almost no JSdoc on the exported functions (and even one empty JSdoc).

Moreover, while I believe that internal documentation should receive as much effort as public documentation (as demonstrated by my PR), I think these two types of documentation do not share the same purpose and should not be treated the same. In this specific case, the `TODO:` purpose was : 
1. assist you during the major update.
2. indicate that it's just a temporary function, which probably should not be used outside its current reference, and definitely should not be used without understanding the context of its inception. That's why I was OK with it showing in the IDE.

But no worries, I have removed the `TODO:` from the JSdoc and took the opportunity to refine the comment a bit. There's no reason to spend more time on this detail, let's move forward 👍

Reply:
If a public function was planned to be removed in a future version, we would publicly document it, like deprecations, otherwise you risk leading users down the wrong path for the future.

As such, in my mind the planned impending removal of something is definitely a relevant part to include in the contract you show to the user, especially when the user here are the other developers of the project. (Of course, in a public function it'd be written more eloquently than a TODO, but this is fine for internal code)

Anyway, this is the nittiest of nit at this point, and not really a good usage of the time of anyone involved 😅 If we feel strongly about this, we can change it later ourselves.